packages/langium/src/serializer/hydrator.ts

```diff
@@ -198,15 +214,15 @@ export class DefaultHydrator implements Hydrator {
                 astNode[name] = arr;
                 for (const item of value) {
```

Comment:
The code between Line 212 until 229 looks duplicated. You may find a solution in the shape of a helper function.

Reply:
We can perform the cleanup in another PR.